### PR TITLE
[cli] Pull in utils changes from #16160

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Split out utils changes from https://github.com/expo/expo/pull/16160 -- not all utils have been tested since they require usage.
- Drops `$EXPO_EDITOR` environment variable in favor of built-in `$EDITOR` variable.
- Pending https://github.com/expo/expo/pull/16507

# How

- Pull in all `utils` code from https://github.com/expo/expo/pull/16160

# Test Plan

Tests should continue to pass.

### CODE REVIEW

File: packages/expo/cli/utils/scheme.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  const { exp } = getConfig(projectRoot);
+  try {
+    const getDefaultScheme = require(resolveFrom(projectRoot, 'expo-dev-client/getDefaultScheme'));
+    const scheme = getDefaultScheme(exp);
+    return scheme;
+  } catch (error) {
+    Log.warn(
+      '\nDevelopment build: Unable to get the default URI scheme for the project. Please make sure the expo-dev-client package is installed.'
+    );
+    // throw new CommandError(error);
```
Comment (EvanBacon) [inline packages/expo/cli/utils/scheme.ts:56]:
```suggestion
```

File: packages/expo/cli/utils/__tests__/editor-test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      .mockImplementationOnce(() => {
+        throw new Error('Could not guess default editor');
+      });
+    guessEditor();
+    expect(editors.getEditor).toBeCalledWith('vscode');
+  });
+});
+
+describe(openInEditorAsync, () => {
+  it(`failed to open in given editor that doesn't exist`, async () => {
```
Comment (EvanBacon) [inline packages/expo/cli/utils/__tests__/editor-test.ts:24]:
```suggestion
  it(`fails to open in a given editor that does not exist`, async () => {
```

File: packages/expo/cli/utils/editor.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        `Failed to auto open path in editor (path: ${path}, binary: ${editor.binary}, preferred: ${inputEditor}):`,
+        error
+      );
+    }
+  }
+
+  if (inputEditor) {
+    Log.error(
+      // TODO: Is this still in use?
+      `Could not resolve editor from environment variable $EXPO_EDITOR="${inputEditor}". Trying again with system default.`
```
Comment (EvanBacon) [inline packages/expo/cli/utils/editor.ts:35]: @byCedric should we keep `$EXPO_EDITOR` around or just utilize the built-in `$EDITOR` variable?

File: packages/expo/cli/utils/getRunningProcess.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+/** Returns a command like `node /Users/evanbacon/.../bin/expo start` or the package.json name. */
+function getProcessCommand(pid: number, procDirectory: string): string {
+  const name = getPackageName(procDirectory);
+
+  if (name) {
+    return name;
+  }
+  return execSync(`ps -o command -p ${pid} | sed -n 2p`, defaultOptions).replace(/\n$/, '').trim();
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo/cli/utils/getRunningProcess.ts:38]: This is from [expo-cli](https://github.com/expo/expo-cli/blob/e5eed5530dafa1599fc74fd50b9e72230bc8093d/packages/xdl/src/utils/getRunningProcess.ts#L25).

File: packages/expo/cli/utils/editor.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    );
+    return openInEditorAsync(path);
+  }
+
+  Log.error(
+    'Could not open editor, you can set it by defining the $EDITOR environment variable with the binary of your editor. (e.g. "code" or "atom")'
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo/cli/utils/editor.ts:41]: This is used inside of [env-editor](https://github.com/sindresorhus/env-editor/blob/bde58471978bd5888ad42e82cd935e8cce1381a7/index.js#L186) which is why I asked @byCedric if we could drop $EXPO_EDITOR.

File: packages/expo/cli/utils/env.ts [hunk 6]
```diff
@@ -21,14 +21,23 @@ export const EXPO_LOCAL = boolish('EXPO_LOCAL', false);
 /** Is running in non-interactive CI mode */
 export const CI = boolish('CI', false);
 
+/** Disable auto web setup */
+export const EXPO_NO_WEB_SETUP = () => boolish('EXPO_NO_WEB_SETUP', false);
+
+/** Disable auto TypeScript setup */
+export const EXPO_NO_TYPESCRIPT_SETUP = () => boolish('EXPO_NO_TYPESCRIPT_SETUP', false);
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo/cli/utils/env.ts]: Artifacts from the split PRs.

File: packages/expo/cli/utils/editor.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (kbrandwijk) [reply packages/expo/cli/utils/editor.ts:41]: Sorry, only saw that comment after I finished the review, it took a while...

**Review by kbrandwijk — APPROVED** [review]

File: packages/expo/cli/utils/editor.ts (hunk 3 above)
Comment (byCedric) [reply packages/expo/cli/utils/editor.ts:35]: I don't think we are using `$EXPO_EDITOR` anymore, [`env-editor` is actually using `$EDITOR`](https://github.com/sindresorhus/env-editor/blob/v0.5.0/index.js#L183). If I recall, it was just for people to pick a different editor for Expo-related stuff. But I think just using `EDITOR` is actually better.

**Pull request merged**